lisp/ess-inf.el

```diff
@@ -101,8 +101,8 @@ been initialised."
         (t
          (progn
            (setq-local inferior-ess-mode-syntax-table
-                       (eval (or (alist-get 'inferior-ess-mode-syntax-table ess-customize-alist)
-                                 (alist-get 'ess-mode-syntax-table ess-customize-alist))))
+                       (eval (or (alist-get 'inferior-ess-mode-syntax-table ess-local-customize-alist)
```

Comment:
I think this would be cleaner if `inferior-ess--set-major-mode` took `cust-alist` as argument.

Reply:
I'm avoiding using the customize-alists too much since our eventual goal is to remove them.

Really, this last part should never happen. If it does we should implement a proper mode.